pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -100,23 +100,34 @@ func (p *singleNumaNodePolicy) getHintMatch(allResourcesHints [][]TopologyHint)
 	return foundMatch, match
 }
 
-// Return hints that have valid bitmasks with exactly one bit set
-func (p *singleNumaNodePolicy) filterHints(allResourcesHints [][]TopologyHint) [][]TopologyHint {
+// Return hints that have valid bitmasks with exactly one bit set. Also return bool
+// which indicates whether allResourceHints only consists of {nil true} hints.
+func (p *singleNumaNodePolicy) filterHints(allResourcesHints [][]TopologyHint) ([][]TopologyHint, bool) {
 	var filteredResourcesHints [][]TopologyHint
+	var noAffinityPreferredHints int
+	var totalHints int
 	if len(allResourcesHints) > 0 {
 		for _, oneResourceHints := range allResourcesHints {
 			var filtered []TopologyHint
 			if len(oneResourceHints) > 0 {
 				for _, hint := range oneResourceHints {
-					if hint.NUMANodeAffinity != nil && hint.NUMANodeAffinity.Count() == 1 {
+					totalHints++
+					if hint.NUMANodeAffinity != nil && hint.NUMANodeAffinity.Count() == 1 && hint.Preferred == true {
 						filtered = append(filtered, hint)
 					}
+					if hint.NUMANodeAffinity == nil && hint.Preferred == true {
+						noAffinityPreferredHints++
+					}
 				}
 			}
 			filteredResourcesHints = append(filteredResourcesHints, filtered)
 		}
 	}
-	return filteredResourcesHints
+	// Check if all resource hints only consist of nil-affinity/preferred hint: {nil true}.
+	if noAffinityPreferredHints == totalHints {
+		return filteredResourcesHints, true
+	}
+	return filteredResourcesHints, false
```

Comment:
As discussed offlline, I think there should be better way to handle this.

Reply:
Agreed, but it does feel like something that should be handled in a `filterHints` function :) - Probably need to have another think about this.

Reply:
This bug fix is now removed and can be dealt with in a future commit.
One option is to do a check just before `filterHints` is called to see if `allResourceHints` consists **only** of `{nil, true}` hints. If so, return `{nil, true}`:
``````
duplicate := make([][]TopologyHint, len(allResourcesHints))
for i := range allResourcesHints {
	duplicate[i] = make([]TopologyHint, len(allResourcesHints[i]))
        for j := range duplicate[i] {
        	duplicate[i][j] = TopologyHint{nil, true}
        }
}

if reflect.DeepEqual(duplicate, allResourcesHints) {
        return TopologyHint{nil, true}
}
``````

Reply:
Is there a test case that fails because we omit this bug fix?

Reply:
Yes, as follows:
```
Single TopologyHint with Preferred as true and NUMANodeAffinity as nil: 
Expected Topology Hint to be {<nil> true}, got {<nil> false}:
```
The problem being that `filterHints` treats hints `{<nil> true}` and `{<nil> false}` equally by appending a nil `[]TopologyHint` in both cases. 

So in both cases they end up returning  `{<nil> false}` when checked after `filterHints` is called. A bit of a corner case but probably one we still need to catch?